package.json

```diff
@@ -2,14 +2,24 @@
   "name": "classnames",
   "version": "2.3.1",
   "description": "A simple utility for conditionally joining classNames together",
+  "type": "module",
+  "sideEffects": false,
   "main": "index.js",
+  "exports": {
+    ".": "./index.js",
+    "./bind": "./bind.js",
+    "./dedupe": "./dedupe.js"
+  },
   "author": "Jed Watson",
   "license": "MIT",
   "repository": {
     "type": "git",
     "url": "https://github.com/JedWatson/classnames.git"
   },
   "types": "./index.d.ts",
+  "engines": {
+    "node": ">=12.22.0"
```

Comment:
According to the Node 12 [changelog](https://github.com/nodejs/node/blob/master/doc/changelogs/CHANGELOG_V12.md#2021-03-30-version-12220-erbium-lts-richardlau) ES modules are considered stable since version 12.22.0, so if we were to use this syntax should we not be using `^12.22`? The exact same reference is present in the Node 14 [changelog](https://github.com/nodejs/node/blob/master/doc/changelogs/CHANGELOG_V14.md#2021-05-11-version-14170-fermium-lts-danielleadams), referencing this [commit](https://github.com/nodejs/node/commit/c1af59384d)  for version 14.17.0.

I can't find references to it being stable before that, but this could simply be a docs change and the code could have stabilized before. If these exist could you point them out?

Reply:
> I can't find references to it being stable before that, but this could simply be a docs change and the code could have stabilized before.

That's right, marking ESM as "stable" in v12.22.0 was just a docs update:

https://github.com/nodejs/node/pull/35781/files

A rationalization for why I use precisely `^12.20 || >= 14.13` in my own packages would take a lengthy blog post, relating to specific aspects of Node.js ESM and package `exports` functionality, the dev dependencies that I use and their level of support, and the way certain commonjs dependencies are structured (e.g. in tests of my packages doing named imports from a `graphql` peer/dev dependency, I rely on Node.js support for detection of CommonJS named exports that was only introduced in v12.20.0, v14.13.0).

My original warning that the range `>=12.22.0` is incorrect was more referring to it allowing v13.x versions that are not compatible.

Your PR is not using package `exports` subpath patterns, but if it does in the future, or if any dependencies or dev dependencies do (very likely) then you need to account for it only being added in v12.20.0, v14.13.0:

https://nodejs.org/api/packages.html#packages_subpath_patterns

Reply:
This makes sense to me, I'll add the version range of `^12.20 || >= 14.13` since that seems to be the lowest and still compatible one.